### ISSUE TITLE
Accessing client info in TcpIn msg

### ISSUE THREAD

**Issue by vicary** [node-red/node-red#1236]
Is there currently a way to access client side info like `msg.req` from HTTP In? Mostly I want client.remote* properties.

**Comment by dceejay** [comment]
Hi,
we prefer general questions on the mailing list or slack channel so we can keep the issue tracker for bugs and code enhancements.
The `msg.req` property is available to any following node - eg function, change, switch etc - you can then get to msg.req.{whatever} as per normal javascript style.

**Comment by vicary** [comment]
Please let me rephrase the issue.

The `TcpIn` node does offer remote client information when in listen mode, see [31-tcpin.js:146](https://github.com/node-red/node-red/blob/master/nodes/core/io/31-tcpin.js#L146).

But somehow other messages in server mode does not show the consistency, see [31-tcpin.js:152](https://github.com/node-red/node-red/blob/master/nodes/core/io/31-tcpin.js#L152), [31-tcpin.js:167](https://github.com/node-red/node-red/blob/master/nodes/core/io/31-tcpin.js#L167).

Users could really use the transparency, for what they expect with similar settings.

**Comment by dceejay** [comment]
Ah - much clearer. yes - they should be consistent.

**Comment by vicary** [comment]
Thank you!
